packages/twenty-server/src/workspace/workspace-sync-metadata/comparators/workspace-field.comparator.ts

```diff
@@ -61,7 +64,8 @@ export class WorkspaceFieldComparator {
         },
         propertiesToStringify: fieldPropertiesToStringify,
         keyFactory(datum) {
-          return datum.name;
+          // Happen when the field is custom
```

Comment:
Maybe we want to check it then?
```
return !datum.isCustom ? datum.standardId : datum.name
```
Or even throw if not present?

Reply:
Yes we can, it's just that `standardId` is `string | null` and `keyExtractor` doesn't accept null value.
`null` should never happen here, but we need it for custom objects, we can put `!` to say it can't be null but I don't really like it.
I go that way so it's less verbose

Reply:
I see, lgtm